### PR TITLE
Allow multi qubit to builder measure

### LINKED ISSUE #4827: pulse builder measure cannot handle multiple qubits
### Information

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?

A code inserts multiple `pulse.measure` instruction creates strange output.

image

### Steps to reproduce the problem

```python
from qiskit import pulse

with pulse.build(backend) as sched:
    with pulse.align_left():
        for i in range(5):
            pulse.measure(i)
```

### What is the expected behavior?

All measure and acquire instructions are aligned in parallel.

### Suggested solutions

This is a defect arising from the constraints of IQX backends measurement operation. Currently it inserts the acquire instruction for all qubits while a measurement pulse is inserted for the specific qubit. Thus repeating `pulse.measure` occupies timeslots of other acquire channels and they cannot be aligned. 

This issue will be automatically solved when the acquisition for single qubit is supported by backend, or we can temporarily solve  this by allowing iterator as `qubit` argument.

### PR DESCRIPTION
**Pull request by nkanazawa1989**
### Summary
fix #4827 

### Details and comments
`measure` returns a register object instead of list of registers when the input is not an iterator. This is for compatibility with existing code (there could be many jupyter execute and documents related to this).

I don't include reno because the builder is not released.

### CODE REVIEW

File: qiskit/pulse/builder.py [hunk 1]
```diff
@@ -1686,25 +1688,32 @@ def measure(qubit: int,
...
+        registers: Register to store result in. If not selected the current
             behaviour is to return the :class:`MemorySlot` with the same
             index as ``qubit``. This register will be returned.
     Returns:
         The ``register`` the qubit measurement result will be stored in.
     """
     backend = active_backend()
-    if not register:
-        register = chans.MemorySlot(qubit)
+    if isinstance(qubits, int):
```
Comment (taalexander) [inline qiskit/pulse/builder.py:1699]: The best way to cast to a list is to try:
```python
try:
    qubits = iter(qubits)
except TypeError:
   qubits = [qubits]

File: qiskit/pulse/builder.py [hunk 2]
```diff
@@ -1636,9 +1636,11 @@ def barrier(*channels_or_qubits: Union[chans.Channel, int]):
 
 
 # Macros
-def measure(qubit: int,
-            register: Union[chans.MemorySlot, chans.RegisterSlot] = None,
-            ) -> Union[chans.MemorySlot, chans.RegisterSlot]:
+def measure(qubits: Union[List[int], int],
+            registers: Union[List[Union[chans.MemorySlot, chans.RegisterSlot]],
```
Comment (taalexander) [inline qiskit/pulse/builder.py:1640]: These types are getting excessively long. You could define a new type above. https://docs.python.org/3/library/typing.html#type-aliases

Comment (nkanazawa1989) [reply qiskit/pulse/builder.py]: done. any suggestion for type name? I defined `MeasResource` to express register + memory slot.

Comment (taalexander) [reply qiskit/pulse/builder.py]: `MeasLoc`, `MeasurementLocation`, `StorageLocation` ?

Comment (nkanazawa1989) [reply qiskit/pulse/builder.py]: `StorageLocation` is much better. thanks

**Pull request merged**